README.md

```diff
@@ -2,20 +2,60 @@
 
 [![Build Status](https://travis-ci.org/bloomberg/clangmetatool.svg?branch=master)](https://travis-ci.org/bloomberg/clangmetatool)
 
-The point of this library is to establish a framework for reusing code
-in clang tools.
+## About clangmetatool
 
-Source control at [https://github.com/bloomberg/clangmetatool/](https://github.com/bloomberg/clangmetatool/)
+When we first started writing clang tools, we realized that there is a
+lot of life cycle management that we had to repeat. In some cases,
+people advocate the usage of global variables to manage the lifecycle
+of that data, but that makes code reuse across tools even harder.
 
-## Building
+Additionally, we also learned that when writing a tool, it will be
+benefitial if the code is split in two phases. First a data collection
+phase, and later a post-processing phase that actually performed the
+bulk of the logic of the tool.
 
-````bash
-mkdir build
-cd build
-cmake -DClang_DIR=/path/to/clang/cmake ..
-make
-make install
-````
+Essentially you will only need to write a class like:
+
+```
```

Comment:
Can you mark this as C++ here so the code get's syntax higlighting?

Reply:
fixed